python/shotgun_fields/checkbox_widget.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Widget that represents the value of a checkbox field in Shotgun
+"""
+from sgtk.platform.qt import QtCore, QtGui
+from .shotgun_field_manager import ShotgunFieldManager
+
+
+class CheckBoxWidget(QtGui.QCheckBox):
+    """
+    Inherited from a :class:`~PySide.QtGui.QCheckBox`, this class is able to
+    display a checkbox field value as returned by the Shotgun API.
+    """
+
+    def __init__(self, parent=None, entity=None, field_name=None, bg_task_manager=None, **kwargs):
```

Comment:
Might should be *args, **kwargs in case positional arguments are used across the board and include additional QCheckBox arguments.